test/unit/matrix/test_matrix.cpp

```diff
@@ -1135,90 +1228,189 @@ TYPED_TEST(MatrixTest, GPUCopy) {
 // already finished (and in case of failure it may not be presented correctly)
 //
 // Note 2:
-// WAIT_GUARD is the time to wait in the launched task for assuring that Matrix d'tor has been called
+// wait_guard is the time to wait in the launched task for assuring that Matrix d'tor has been called
 // after going out-of-scope. This duration must be kept as low as possible in order to not waste time
 // during tests, but at the same time it must be enough to let the "main" to arrive to the end of the
 // scope.
+//
+// Note 3:
+// The tests about lifetime of a Matrix built with user provided memory are not examples of good
+// usage, but they are just meant to test that the Matrix does not wait on destruction for any left
+// task on one of its tiles.
 
-const auto WAIT_GUARD = std::chrono::milliseconds(10);
+const auto wait_guard = std::chrono::milliseconds(100);
 const auto device = dlaf::Device::CPU;
 using TypeParam = std::complex<float>;  // randomly chosen element type for matrix
 
+// Create a single-element matrix
 template <class T>
 auto createMatrix() -> Matrix<T, device> {
   return {{1, 1}, {1, 1}};
 }
 
+// Create a single-element matrix with user-provided memory
 template <class T>
-auto createConstMatrix() {
-  LayoutInfo layout({1, 1}, {1, 1}, 1, 1, 1);
-  memory::MemoryView<T, device> mem(layout.minMemSize());
-  const T* p = mem();
+auto createMatrix(T& data) -> Matrix<T, device> {
+  return createMatrixFromColMajor<Device::CPU>({1, 1}, {1, 1}, 1, &data);
+}
 
-  return Matrix<const T, device>{layout, p};
+// Create a single-element const matrix with user-provided memory
+template <class T>
+auto createConstMatrix(const T& data) {
+  return createMatrixFromColMajor<Device::CPU>({1, 1}, {1, 1}, 1, &data);
 }
 
 TEST(MatrixDestructorFutures, NonConstAfterRead) {
   hpx::future<void> last_task;
 
-  std::atomic<bool> guard{false};
+  std::atomic<bool> is_exited_from_scope{false};
   {
     auto matrix = createMatrix<TypeParam>();
 
     auto shared_future = matrix.read(LocalTileIndex(0, 0));
-    last_task = shared_future.then(hpx::launch::async, [&guard](auto&&) {
-      hpx::this_thread::sleep_for(WAIT_GUARD);
-      EXPECT_FALSE(guard);
+    last_task = shared_future.then(hpx::launch::async, [&is_exited_from_scope](auto&&) {
+      hpx::this_thread::sleep_for(wait_guard);
+      EXPECT_TRUE(is_exited_from_scope);
```

Comment:
I'm wondering if giving more time (e.g. 2 seconds) wouldn't be even better:
(I suggest to use `wait_guard = std::chrono::milliseconds(20);`)
```suggestion
      for (int i=0; i < 10 && !is_exited_from_scope; ++i) {
        hpx::this_thread::sleep_for(wait_guard);
      }
      EXPECT_TRUE(is_exited_from_scope);
```

Reply:
Ok, so you're proposal is to wait the least time possible with a granularity defined by `wait_guard`, up until a maximum duration which is eventually way longer than the currently extended one.

I'll do that.

Reply:
See if the change here 136ea48ec12729304eee62d86169680491002b76 is ok for you.

Moreover, pay attention to the last change of the commit, where I extended the time for the only guard that works differently from the others.